streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -198,9 +199,10 @@ public synchronized void close() {
             state = STOPPED;
 
             log.info("Stopped Kafka Stream process");
-        } else {
-            throw new IllegalStateException("This process has not started yet.");
         }
+
+        metrics.close();
```

Comment:
We no longer throw an exception here, and we've moved the `metrics.close` call outside the conditional so that a call to `KafkaStreams.close` will close the Metrics instance whether `start` has been called or not.